### ISSUE TITLE
Webhook is not actionable

### ISSUE THREAD

**Issue by bogdans92** [Hamuko/anifunnel#3]
Hello again,

I started receiving the below mentioned errors, webhook is not actionable.

Initially i thought it was it because the token for Anilist expired so i regenerated it and created a new image with the new token.

Have you ran into this issue by chance?

Plex version: Version 4.108.0

2024-01-14 14:30:54 2024-01-14T12:30:54.821Z INFO  [rocket::server::_] Matched: (scrobble) POST /
2024-01-14 14:30:54 2024-01-14T12:30:54.821Z INFO  [anifunnel] Webhook is not actionable
2024-01-14 14:30:54 2024-01-14T12:30:54.822Z INFO  [rocket::server::_] Outcome: Success(200 OK)
2024-01-14 14:30:54 2024-01-14T12:30:54.822Z INFO  [rocket::server::_] Response succeeded.
2024-01-14 14:31:04 2024-01-14T12:31:04.251Z INFO  [rocket::server] POST / multipart/form-data:
2024-01-14 14:31:04 2024-01-14T12:31:04.251Z INFO  [rocket::server::_] Matched: (scrobble) POST /
2024-01-14 14:31:04 2024-01-14T12:31:04.251Z INFO  [anifunnel] Webhook is not actionable
2024-01-14 14:31:04 2024-01-14T12:31:04.251Z INFO  [rocket::server::_] Outcome: Success(200 OK)
2024-01-14 14:31:04 2024-01-14T12:31:04.251Z INFO  [rocket::server::_] Response succeeded.
2024-01-14 14:31:10 2024-01-14T12:31:10.198Z INFO  [rocket::server] POST / multipart/form-data:
2024-01-14 14:31:10 2024-01-14T12:31:10.199Z INFO  [rocket::server::_] Matched: (scrobble) POST /
2024-01-14 14:31:10 2024-01-14T12:31:10.199Z INFO  [anifunnel] Webhook is not actionable
2024-01-14 14:31:10 2024-01-14T12:31:10.199Z INFO  [rocket::server::_] Outcome: Success(200 OK)
2024-01-14 14:31:10 2024-01-14T12:31:10.199Z INFO  [rocket::server::_] Response succeeded.

Thank you,

**Comment by Hamuko** [comment]
"Webhook is not actionable" means one of the three following things:

1. **It's not a "media.scrobble" webhook event.** Plex doesn't filter what it sends to anifunnel, so anifunnel also gets webhooks for new media being added to the library, playback being started, etc. Since anifunnel only updates what you've watched, it's only watching for "media.scrobble" events, and everything else is non-actionable.
2. **The scrobbled media is not an "episode" type.** Plex also sends media.scrobble for movies and music, and anifunnel doesn't handle those, therefore those webhooks are non-actionable.
3. **The watched episode is from a season higher than 1 and multi-season support isn't enabled.** By default, anifunnel doesn't handle multi-season shows. See the "Multi-season shows" section of the README.

Basically, it's very normal to see these in your logs and should only be an issue if you have anime in your library with multiple seasons.

**Comment by bogdans92** [comment]
Hello,

Thank you for the insights. The anime is not from a second season, i'll keep on monitoring and update.

Q: is there a possibility to pass the multi season argument to the docker image when creating it?

Not sure what the issues is.

Thanks

**Comment by Hamuko** [comment]
Use the `ANIFUNNEL_MULTI_SEASON` environment variable.

**Comment by bogdans92** [comment]
Thanks, that works.

I also confirmed that the webhook is working and that the Info message is just that.

Q; When does the anime get updated? Any specific time that it needs to trigger?

Any plans on adding the following features?
1. Display match log for when anime starts (e.g when you start playing an anime to log that it matched from the list)
2. Display in how long the anime progress will be updated
3. Configure after how many minutes of watching the anime list will be updated.

Thank you.

**Comment by Hamuko** [comment]
>Display match log for when anime starts (e.g when you start playing an anime to log that it matched from the list)

Haven't thought about it but it might be something worth doing at some point. Not really a high priority though.

>Display in how long the anime progress will be updated

It'll happen when you've watched 90% of the given episode. So for your typical 24-minute anime, the scrobble will happen at 21 minutes and 36 seconds.

>Configure after how many minutes of watching the anime list will be updated.

The 90% mark is when Plex sends the "media.scrobble" event, so it's not configurable on anifunnel's side, or even on Plex's side.

**Comment by bogdans92** [comment]
Got it. Thank you very much for all the hard work !
